src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
Done, and it passes the `IdentityStrategy*` tests, but fails this one:
```
Failed tests: 
  CollectionItemElementSelectorTest.should have same hashCode as item:46 Condition not satisfied:

element.hashCode() == item.hashCode()
|       |          |  |    |
[foo]   0          |  foo  101574
                   false
```
Did not commit it now, will send you the patch through Slack chat.